dot/parachain/availability-store/availability_store_test.go

```diff
@@ -1752,3 +1795,144 @@ func TestStoredDataKeptUntilFinalized(t *testing.T) {
 	err = harness.overseer.Stop()
 	require.NoError(t, err)
 }
+
+func TestForkfullnessWorks(t *testing.T) {
+	harness := newTestHarness(t)
+	nValidators := uint32(10)
+	msgSenderChan := make(chan error)
+
+	blockNumber1 := uint32(5)
+	parent1 := common.Hash{0x03, 0x03, 0x03, 0x03}
+	pov1 := parachaintypes.PoV{BlockData: parachaintypes.BlockData{1, 2, 3}}
+	pov1Hash := common.MustBlake2bHash(scale.MustMarshal(pov1))
+	candidate1 := buildCandidateReceipt(pov1Hash)
+	candidate1Hash, err := candidate1.Hash()
+	require.NoError(t, err)
+	availableData1 := AvailableData{
+		PoV: pov1,
+		ValidationData: parachaintypes.PersistedValidationData{
+			ParentHead: parachaintypes.HeadData{Data: []byte{}},
+		},
+	}
+	availableData1ErasureRoot := buildAvailableDataBranchesRoot(t, nValidators, availableData1)
+	msg1StoreAvailableData := StoreAvailableData{
+		CandidateHash:       parachaintypes.CandidateHash{Value: candidate1Hash},
+		NumValidators:       nValidators,
+		AvailableData:       availableData1,
+		ExpectedErasureRoot: availableData1ErasureRoot,
+		Sender:              msgSenderChan,
+	}
+	candidate1Events, err := parachaintypes.NewCandidateEvents()
+	require.NoError(harness.t, err)
+	candidate1Events.Add(parachaintypes.CandidateIncluded{CandidateReceipt: candidate1})
+
+	blockNumber2 := uint32(5)
+	parent2 := common.Hash{0x04, 0x04, 0x04, 0x04}
+	pov2 := parachaintypes.PoV{BlockData: parachaintypes.BlockData{4, 5, 6}}
+	pov2Hash := common.MustBlake2bHash(scale.MustMarshal(pov2))
+	candidate2 := buildCandidateReceipt(pov2Hash)
+	candidate2Hash, err := candidate2.Hash()
+	require.NoError(t, err)
+	availableData2 := AvailableData{
+		PoV: pov2,
+		ValidationData: parachaintypes.PersistedValidationData{
+			ParentHead: parachaintypes.HeadData{Data: []byte{}},
+		},
+	}
+	availableData2ErasureRoot := buildAvailableDataBranchesRoot(t, nValidators, availableData2)
+	msg2StoreAvailabeData := StoreAvailableData{
+		CandidateHash:       parachaintypes.CandidateHash{Value: candidate2Hash},
+		NumValidators:       nValidators,
+		AvailableData:       availableData2,
+		ExpectedErasureRoot: availableData2ErasureRoot,
+		Sender:              msgSenderChan,
+	}
+	candidate2Events, err := parachaintypes.NewCandidateEvents()
+	require.NoError(harness.t, err)
+	candidate2Events.Add(parachaintypes.CandidateIncluded{CandidateReceipt: candidate2})
+
+	err = harness.overseer.Start()
+	require.NoError(t, err)
+
+	go harness.processMessages()
```

Comment:
What are the chances that this goroutine keeps running even if this test finishes?

Reply:
I've added defer stop to these tests to handle this.